spanner/spanner_snippets/snippet.go

```diff
@@ -906,6 +916,264 @@ func queryWithTimestamp(ctx context.Context, w io.Writer, client *spanner.Client
+
+// [END spanner_dml_write_then_read]
+
+// [START spanner_dml_structs]
+
+func updateUsingDMLStruct(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	_, err := client.ReadWriteTransaction(
+		ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+
+			type nameStruct struct {
+				FirstName string
+				LastName  string
+			}
+			var singerInfo = nameStruct{"Timothy", "Campbell"}
+
+			rowCount, err := txn.Update(ctx, spanner.Statement{
+				SQL: `Update Singers Set LastName = 'Grant' 
+						WHERE STRUCT<FirstName String, LastName String>(Firstname, LastName) = @name`,
+				Params: map[string]interface{}{"name": singerInfo},
+			})
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(w, "%d record(s) inserted.\n", rowCount)
+			return nil
+		})
+	return err
+}
+
+// [END spanner_dml_structs]
+
+// [START spanner_dml_getting_started_insert]
+
+func writeUsingDML(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	_, err := client.ReadWriteTransaction(
+		ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+			rowCount, err := txn.Update(ctx, spanner.Statement{
+				SQL: `INSERT Singers (SingerId, FirstName, LastName) VALUES
+								(12, 'Melissa', 'Garcia'),
+								(13, 'Russell', 'Morales'),
+								(14, 'Jacqueline', 'Long'),
+								(15, 'Dylan', 'Shaw')`})
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(w, "%d record(s) inserted.\n", rowCount)
+			return err
+		})
+	return err
+}
+
+// [END spanner_dml_getting_started_insert]
+
+// [START spanner_dml_getting_started_update]
+
+func writeWithTransactionUsingDML(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	_, err := client.ReadWriteTransaction(
+		ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+			// Helper function for retrieving the budget of a record
+			getBudget := func(key spanner.Key) (int64, error) {
+				row, err := txn.ReadRow(ctx, "Albums", key, []string{"MarketingBudget"})
+				if err != nil {
+					return 0, err
+				}
+				var budget int64
+				if err := row.Column(0, &budget); err != nil {
+					return 0, err
+				}
+				return budget, nil
+			}
+			// Helper function for updating the budget of a record.
+			updateBudget := func(singerId, albumId, albumBudget int64) error {
+				_, err := txn.Update(ctx, spanner.Statement{
+					SQL: `UPDATE Albums
+									SET MarketingBudget = @AlbumBudget
+									WHERE SingerId = @SingerId and AlbumId = @AlbumId`,
+					Params: map[string]interface{}{
+						"SingerId":    singerId,
+						"AlbumId":     albumId,
+						"AlbumBudget": albumBudget,
+					},
+				})
+				return err
+			}
+
+			// Transfer the marketing budget from one album to another. By keeping the actions
+			// in a single transaction, it ensures the movement is atomic.
+			const transferAmt = 200000
+			var album1budget, album2budget int64
+			var err error
+			if album1budget, err = getBudget(spanner.Key{1, 1}); err != nil {
```

Comment:
It's not immediately clear to me where the `spanner.Key{1, 1}` and `spanner.Key{2, 2}` come from. Is there a way to make it more obvious?